packages/gatsby-source-contentful/src/gatsby-node.js

```diff
@@ -121,24 +119,53 @@ exports.sourceNodes = async (
   // are "updated" so will get the now deleted reference removed.
 
   function deleteContentfulNode(node) {
-    const localizedNodes = locales
-      .map(locale => {
-        const nodeId = createNodeId(
-          normalize.makeId({
-            spaceId: space.sys.id,
-            id: node.sys.id,
-            currentLocale: locale.code,
-            defaultLocale,
+    locales.forEach(locale => {
+      const nodeId = createNodeId(
+        normalize.makeId({
+          spaceId: space.sys.id,
+          id: node.sys.id,
+          currentLocale: locale.code,
+          defaultLocale,
+        })
+      )
+      const gatsbyNode = getNode(nodeId)
+
+      if (gatsbyNode) {
+        Object.keys(gatsbyNode).forEach(field => {
+          // we only care about ___NODE as they are links to other ones
+          if (!field.includes(`___NODE`)) {
+            return
+          }
+
+          // remove foreignReferences to this node
+          ;[].concat(gatsbyNode[field]).forEach(value => {
+            const referencedNode = getNode(value)
+
+            if (referencedNode) {
+              const possibleReferences =
+                referencedNode[`${gatsbyNode.parent.toLowerCase()}___NODE`]
+
+              if (
+                possibleReferences &&
+                possibleReferences.indexOf(gatsbyNode.id) > -1
```

Comment:
```suggestion
                possibleReferences.includes(gatsbyNode.id)
```

Reply:
Otoh, if you're going to re-use this value below anyways, you may as well cache the index and prevent a double search. If the size of `possibleReferences` is only ever going to be a <10 number of items then I'd suggest `includes` for legibility.